pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -0,0 +1,1150 @@
+			if l, ok := podResourceLimits[name]; ok {
+				podResourceLimits[name] = l + limit.MilliValue()
+			} else {
+				podResourceLimits[name] = limit.MilliValue()
+			}
+		}
+		for name, req := range c.Resources.Requests {
+			if l, ok := podResourceRequests[name]; ok {
+				podResourceRequests[name] = l + req.MilliValue()
+			} else {
+				podResourceRequests[name] = req.MilliValue()
+			}
+		}
+
+		opts, err := m.runtimeHelper.GenerateRunContainerOptions(pod, &c, "")
+		if err != nil {
+			return nil, err
+		}
+
+		for _, port := range opts.PortMappings {
+			hostPort := int32(port.HostPort)
+			containerPort := int32(port.ContainerPort)
+			protocol := toRuntimeProtocol(port.Protocol)
+			portMappings = append(portMappings, &runtimeApi.PortMapping{
+				HostIp:        &port.HostIP,
+				HostPort:      &hostPort,
+				ContainerPort: &containerPort,
+				Protocol:      &protocol,
+				Name:          &port.Name,
+			})
+		}
+
+		cgroupParent = opts.CgroupParent
+	}
+
+	if cgroupParent != "" {
+		if podSandboxConfig.Linux != nil {
+			podSandboxConfig.Linux.CgroupParent = &cgroupParent
+		} else {
+			podSandboxConfig.Linux = &runtimeApi.LinuxPodSandboxConfig{
+				CgroupParent: &cgroupParent,
+			}
+		}
+	}
+
+	if len(portMappings) > 0 {
+		podSandboxConfig.PortMappings = portMappings
+	}
+
+	if len(podResourceLimits) > 0 || len(podResourceRequests) > 0 {
+		podSandboxConfig.Resources = &runtimeApi.PodSandboxResources{}
+		var podCPULimit, podMemoryLimit, podCPURequest, podMemoryRequest int64
+		for k, v := range podResourceLimits {
+			switch k {
+			case api.ResourceCPU:
+				podCPULimit = v
+			case api.ResourceMemory:
+				podMemoryLimit = v
+			}
+		}
+		for k, v := range podResourceRequests {
+			switch k {
+			case api.ResourceCPU:
+				podCPURequest = v
+			case api.ResourceMemory:
+				podMemoryRequest = v
+			}
+		}
+		if podCPULimit != 0 || podCPURequest != 0 {
+			cpuResource := &runtimeApi.ResourceRequirements{}
+			if podCPULimit != 0 {
+				cpulimit := float64(podCPULimit) / 1000
+				cpuResource.Limits = &cpulimit
+			}
+			if podCPURequest != 0 {
+				cpuRequest := float64(podCPURequest) / 1000
+				cpuResource.Requests = &cpuRequest
+			}
+			podSandboxConfig.Resources.Cpu = cpuResource
+		}
+		if podMemoryLimit != 0 || podMemoryRequest != 0 {
+			memoryResource := &runtimeApi.ResourceRequirements{}
+			if podMemoryLimit != 0 {
+				memLimit := float64(podMemoryLimit) / 1000
+				memoryResource.Limits = &memLimit
+			}
+			if podMemoryRequest != 0 {
+				memRequest := float64(podMemoryRequest) / 1000
+				memoryResource.Requests = &memRequest
+			}
+			podSandboxConfig.Resources.Memory = memoryResource
+		}
+	}
+
+	return podSandboxConfig, nil
+}
+
+func (m *kubeGenericRuntimeManager) createPodSandbox(pod *api.Pod) (string, error, string) {
+	podSandboxConfig, err := m.generatePodSandboxConfig(pod)
+	if err != nil {
+		message := fmt.Sprintf("GeneratePodSandboxConfig failed: %v", err)
+		glog.Error(message)
```

Comment:
In general, we try to use `format.Pod(pod)` in log messages to help identify the pod. Could you change the glog lines to do that?